submit_cgap/submission.py

```diff
@@ -160,6 +161,70 @@ def get_defaulted_project(project, user_record):
     return project
 
 
+def get_defaulted_award(award, user_record):
+    """
+    Returns the given award or else if none is specified, it tries to infer an award.
+
+    :param award: the @id of an award, or None
+    :param user_record: the user record for the authorized user
+    :return: the @id of an award to use
+    """
+
+    if not award:
+        # The lab is expected to have awards looking like:
+        #  [
+        #    {"@id": "/awards/foo", ...},
+        #    {"@id": "/awards/bar", ...},
+        #    {"@id": "/awards/baz", ...},
+        #  ]
+        lab = user_record.get('lab', {})
+        lab_awards = lab.get('awards', [])
+        if len(lab_awards) == 0:
+            raise SyntaxError("Your user profile declares no lab with awards.")
+        elif len(lab_awards) > 1:
+            options = disjoined_list([award['@id'] for award in lab_awards])
+            raise SyntaxError(f"Your lab ({lab['@id']}) declares multiple awards."
+                              f" You must explicitly specify one of {options} with --award.")
+        else:
+            [lab_award] = lab_awards
+            award = lab_award['@id']
+            show("Using award:", award)
+    return award
+
+
+def get_defaulted_lab(lab, user_record):
+    """
+    Returns the given lab or else if none is specified, it tries to infer a lab.
+
+    :param lab: the @id of a lab, or None
+    :param user_record: the user record for the authorized user
+    :return: the @id of a lab to use
+    """
+
+    if not lab:
+        lab = user_record.get('lab', {}).get('@id', None)
+        if not lab:
+            raise SyntaxError("Your user profile has no lab declared,"
+                              " so you must specify --lab explicitly.")
+        show("Using lab:", lab)
+    return lab
+
+
+APP_ARG_DEFAULTERS = {
+    'institution': get_defaulted_institution,
+    'probject': get_defaulted_project,
```

Comment:
Typo introduced somehow here, broke CGAP specific submit, fixed and checked in myself, hope you don't mind.

Reply:
(You fixed this.)